nifi-nar-bundles/nifi-extension-utils/nifi-event-transport/src/main/java/org/apache/nifi/event/transport/netty/NettyEventSender.java

```diff
@@ -0,0 +1,100 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.nifi.event.transport.netty;
+
+import io.netty.channel.Channel;
+import io.netty.channel.ChannelFuture;
+import io.netty.channel.EventLoopGroup;
+import io.netty.channel.pool.ChannelPool;
+import io.netty.util.concurrent.Future;
+import org.apache.nifi.event.transport.EventException;
+import org.apache.nifi.event.transport.EventSender;
+
+import java.net.SocketAddress;
+
+/**
+ * Netty Event Sender with Channel Pool
+ *
+ * @param <T> Event Type
+ */
+class NettyEventSender<T> implements EventSender<T> {
+    private final EventLoopGroup group;
+
+    private final ChannelPool channelPool;
+
+    private final SocketAddress remoteAddress;
+
+    /**
+     * Netty Channel Event Sender with Event Loop Group and Channel Pool
+     *
+     * @param group Event Loop Group
+     * @param channelPool Channel Pool
+     * @param remoteAddress Remote Address
+     */
+    NettyEventSender(final EventLoopGroup group, final ChannelPool channelPool, final SocketAddress remoteAddress) {
+        this.group = group;
+        this.channelPool = channelPool;
+        this.remoteAddress = remoteAddress;
+    }
+
+    /**
+     * Send Event using Channel acquired from Channel Pool
+     *
+     * @param event Event
+     */
+    @Override
+    public void sendEvent(final T event) {
+        try {
+            final Future<Channel> futureChannel = channelPool.acquire().sync();
+            final Channel channel = futureChannel.get();
+            try {
```

Comment:
Can the channel not be active here?  Can it not be writeable?

Reply:
The `FixedChannelPool` configured in `NettyEventSenderFactory` includes a `ChannelHealthChecker` that checks whether the channel is active and connected.  That should minimize the possible types of exceptions thrown in this try block, but the finally block is intended to ensure that the `Channel` is always returned to the pool.